packages/eslint-plugin/tests/rules/prefer-optional-chain/prefer-optional-chain.test.ts

```diff
@@ -665,35 +665,189 @@ describe('|| {}', () => {
+      {
+        code: noFormat`if (foo) foo.bar()`, // Missing semi-colon and curly braces
+        errors: [{ messageId: 'preferOptionalChain', suggestions: null }],
+        options: [{ allowSuggestingOnIfStatements: true }],
+        output: 'foo?.bar()',
+      },
+      {
+        code: noFormat`if (foo) { foo(); }`,
+        errors: [{ messageId: 'preferOptionalChain', suggestions: null }],
+        options: [{ allowSuggestingOnIfStatements: true }],
+        output: 'foo?.();',
+      },
+      {
+        code: noFormat`if (foo) { foo[bar](); }`,
+        errors: [{ messageId: 'preferOptionalChain', suggestions: null }],
+        options: [{ allowSuggestingOnIfStatements: true }],
+        output: 'foo?.[bar]();',
+      },
+      {
+        code: noFormat`if (foo[bar]) { foo[bar].baz(); }`,
+        errors: [{ messageId: 'preferOptionalChain', suggestions: null }],
+        options: [{ allowSuggestingOnIfStatements: true }],
+        output: 'foo[bar]?.baz();',
+      },
+      {
+        code: noFormat`if (foo.bar.baz()) { foo.bar.baz().bazz(); }`,
+        errors: [{ messageId: 'preferOptionalChain', suggestions: null }],
+        options: [{ allowSuggestingOnIfStatements: true }],
+        output: 'foo.bar.baz()?.bazz();',
+      },
+      {
+        code: noFormat`if (foo && foo.bar && foo.bar.baz) { foo.bar.baz.bazz(); }`,
+        errors: [{ messageId: 'preferOptionalChain', suggestions: null }],
+        options: [{ allowSuggestingOnIfStatements: true }],
+        output: 'foo?.bar?.baz?.bazz();',
+      },
+      {
+        code: noFormat`if (foo) { foo.bar.baz && foo.bar.baz(); }`,
+        errors: [{ messageId: 'preferOptionalChain', suggestions: null }],
+        options: [{ allowSuggestingOnIfStatements: true }],
+        output: 'foo?.bar.baz?.();',
+      },
+      {
+        code: `
+          declare const foo: undefined | { bar?: { baz?: { bazz: () => void } } };
+          if (foo && foo.bar && foo.bar.baz) {
+            foo.bar.baz.bazz();
+          }
+        `,
+        errors: [{ messageId: 'preferOptionalChain', suggestions: null }],
+        options: [
+          { allowSuggestingOnIfStatements: true, requireNullish: true },
+        ],
+        output: `
+          declare const foo: undefined | { bar?: { baz?: { bazz: () => void } } };
+          foo?.bar?.baz?.bazz();
+        `,
+      },
+    ],
+    valid: [
+      // Option is disabled
+      {
+        code: noFormat`if (foo) { foo.bar(); }`,
+        options: [{ allowSuggestingOnIfStatements: false }],
+      },
+      // Ignore no calls
+      {
+        code: noFormat`if (foo) { foo.bar; }`,
+        options: [{ allowSuggestingOnIfStatements: true }],
+      },
+      {
+        code: noFormat`if (foo) { foo.bar?.baz; }`,
+        options: [{ allowSuggestingOnIfStatements: true }],
+      },
+      // Ignore when comment exists before or after statement inside consequent
+      {
+        code: noFormat`if (foo) { /* comment */ foo.bar(); }`,
```

Comment:
[Bug] Why ignore this? I would think we'd still want to report, and have the fixer/suggestion move the comment. This `/* comment */` doesn't change the fact that the code violates the rule's new preference.